extensions/ql-vscode/test/vscode-tests/no-workspace/language-selection-panel/language-selection-data-provider.test.ts

```diff
@@ -0,0 +1,66 @@
+import {
+  QueryLanguage,
+  getLanguageDisplayName,
+} from "../../../../src/common/query-language";
+import { LanguageContextStore } from "../../../../src/language-context-store";
+import {
+  LanguageSelectionTreeDataProvider,
+  LanguageSelectionTreeViewItem,
+} from "../../../../src/language-selection-panel/language-selection-data-provider";
+import { createMockApp } from "../../../__mocks__/appMock";
+import { EventEmitter, ThemeIcon } from "vscode";
+
+describe("LanguageSelectionTreeDataProvider", () => {
+  function expectSelected(
+    items: LanguageSelectionTreeViewItem[],
+    expected: QueryLanguage | undefined,
+  ) {
+    items.forEach((item) => {
+      if (item.language === expected) {
+        expect(item.selected).toBe(true);
+        expect(item.iconPath).toEqual(new ThemeIcon("check"));
+      } else {
+        expect(item.selected).toBe(false);
+        expect(item.iconPath).toBe(undefined);
+      }
+    });
+  }
+
+  describe("getChildren", () => {
+    const app = createMockApp({
+      createEventEmitter: <T>() => new EventEmitter<T>(),
+    });
+    const languageContext = new LanguageContextStore(app);
+    const dataProvider = new LanguageSelectionTreeDataProvider(languageContext);
+
+    it("returns list of all languages", async () => {
+      const expectedLanguageNames = [
+        "All languages",
+        ...Object.values(QueryLanguage).map((language) => {
+          return getLanguageDisplayName(language);
+        }),
+      ];
+      // Note that the internal order of C# and C / C++ is different from what is shown in the UI.
+      [expectedLanguageNames[1], expectedLanguageNames[2]] = [
+        expectedLanguageNames[2],
+        expectedLanguageNames[1],
+      ];
```

Comment:
Do we want to test this exact order or would it also be fine to say that `actualLanguageNames.sort()` is equal to `expectedLanguageNames.sort()`?

Reply:
I wasnt super sure. The easiest would just be to match the internal order of the elements, but I wasnt sure whether maintaining the specific order was required.
I can add the sort for now, and we can always tweak it.